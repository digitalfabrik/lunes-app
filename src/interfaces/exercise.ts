export interface IVocabularyOverviewScreen {
  navigation: any
  route: any
}

export interface IVocabularyTrainerScreen {
  navigation: any
  route: any
}

export interface IAlternativeWordProps {
  alt_word: string
  article: string
}

export interface IDocumentProps {
  id: number
  word: string
  article: string
  document_image: Array<{ id: number; image: string }>
  audio: string
  alternatives: IAlternativeWordProps[]
}

export interface IVocabularyOverviewListItemProps {
  id: number
  word: string
  article: string
  image: string
  audio: string
}

export interface IConfirmationModalProps {
  visible: boolean
  setIsModalVisible: Function
  navigation: any
  extraParams: any
}

export interface IAnswerSectionProps {
  tryLater: () => void
  currentDocumentNumber: number
  setCurrentDocumentNumber: Function
  documents: IDocumentProps[]
  finishExercise: Function
  trainingSet: string
  disciplineTitle: string
}

export interface IFeedbackProps {
  secondAttempt: boolean
  result: string
  document?: IDocumentProps
  input: string
}

export interface IActionsProps {
  tryLater: () => void
  giveUp: () => void
  result: string
  checkEntry: () => void
  getNextWord: () => void
  input: string
  isFinished: boolean
  secondAttempt: boolean
}
