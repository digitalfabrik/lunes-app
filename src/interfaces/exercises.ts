export interface IExercisesScreenProps {
  route: any
  navigation: any
}
