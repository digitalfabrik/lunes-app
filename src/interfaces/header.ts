export interface IHeaderProps {
  top: number | undefined
}
