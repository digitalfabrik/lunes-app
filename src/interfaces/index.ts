import { ReactElement } from 'react'

export interface ILoadingProps {
  children: ReactElement
  isLoading: boolean
}

export interface IButtonProps {
  onPress: () => void
  disabled?: boolean
  children: ReactElement
  theme?: string
  testID?: string
}
export interface IPopoverProps {
  setIsPopoverVisible: (param: boolean) => void
  isVisible: boolean
  children: ReactElement
}
