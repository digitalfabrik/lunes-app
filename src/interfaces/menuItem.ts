import { ReactElement } from 'react'

export interface IMenuItemProps {
  selected: boolean
  onPress: () => void
  icon: string
  title: string
  children: ReactElement
}
