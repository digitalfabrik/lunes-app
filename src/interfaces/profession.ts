export interface IProfessionScreenProps {
  navigation: any
}

export interface IProfessionsProps {
  id: number
  title: string
  description: string
  icon?: string
  total_training_sets: number
}
