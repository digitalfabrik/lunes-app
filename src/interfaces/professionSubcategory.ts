export interface IProfessionSubcategoryScreenProps {
  route: any
  navigation: any
}

export interface IProfessionSubcategoryProps {
  id: number
  title: string
  description: string
  icon: string
  total_documents: number
}
