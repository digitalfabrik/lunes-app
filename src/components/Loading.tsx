import React from 'react'
import { View, ActivityIndicator, StyleSheet } from 'react-native'
import { COLORS } from '../constants/colors'
import { ILoadingProps } from '../interfaces'

export const styles = StyleSheet.create({
  root: {
    flex: 1,
    alignItems: 'center'
  },
  indicator: {
    height: '50%',
    justifyContent: 'center',
    alignItems: 'center'
  }
})
const Loading = ({ children, isLoading }: ILoadingProps) => (
  <View style={styles.root}>
    {isLoading ? (
      <View style={styles.indicator}>
        <ActivityIndicator size='large' color={COLORS.lunesBlack} />
      </View>
    ) : (
      children
    )}
  </View>
)

export default Loading
