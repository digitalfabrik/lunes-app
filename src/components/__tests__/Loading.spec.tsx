import { Text } from 'react-native'
import React from 'react'
import Loading from '../Loading'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { ILoadingProps } from '../../interfaces'

describe('Components', () => {
  describe('Loading ', () => {
    const defaultLoadingProps: ILoadingProps = {
      isLoading: true,
      children: <Text>Test Children</Text>
    }

    it('renders correctly across screens', () => {
      const component = shallow(<Loading {...defaultLoadingProps} />)
      expect(toJson(component)).toMatchSnapshot()
    })

    it('should not render children when isLoading is true', () => {
      const loadingProps: ILoadingProps = {
        ...defaultLoadingProps
      }

      const component = shallow(<Loading {...loadingProps} />)
      expect(component.contains(loadingProps.children)).toBe(false)
    })

    it('should render children when isLoading is false', () => {
      const loadingProps: ILoadingProps = {
        ...defaultLoadingProps,
        isLoading: false
      }

      const component = shallow(<Loading {...loadingProps} />)
      expect(component.contains(loadingProps.children)).toBe(true)
    })
  })
})
