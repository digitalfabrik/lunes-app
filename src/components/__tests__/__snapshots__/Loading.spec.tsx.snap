// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Loading  renders correctly across screens 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": "50%",
        "justifyContent": "center",
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#000e38"
      hidesWhenStopped={true}
      size="large"
    />
  </View>
</View>
`;
